cl/scrapers/utils.py

```diff
@@ -446,3 +449,37 @@ def scraped_citation_object_is_valid(citation_object: dict) -> bool:
         logger.error("Parsed reporter '%s' does not exist", parsed_reporter)
 
     return False
+
+
+def save_response(site: AbstractSite) -> None:
+    """Stores scrapers responses content and headers in a S3 bucket
+
+    This is passed to juriscraper's Site objects as the
+    `save_response_fn` argument, which will make Juriscraper
+    save every response
+
+    :param site: the Site object, used to access the saved response
+    :return None
+    """
+
+    storage = S3GlacierInstantRetrievalStorage()
+    response = site.request["response"]
+
+    scraper_id = site.court_id.split(".")[-1]
+    scrape_type = site.court_id.split(".")[1]  # opinions or oral args
+    now_str = datetime.now().strftime("%Y/%m/%d/%H_%M_%S")
+    base_name = f"responses/{scrape_type}/{scraper_id}/{now_str}"
+
+    headers_json = json.dumps(dict(response.headers), indent=4)
+    storage.save(f"{base_name}_headers.json", ContentFile(headers_json))
+
+    try:
+        # both tests for and parses JSON content
+        content = json.loads(response.content)
+        extension = "json"
+    except:
```

Comment:
Definitely need the exception type here, but otherwise this looks great and ready to merge to me.

Reply:
@mlissner I added the exception types, I think it should be ready to merge